### PR TITLE
[FrameworkBundle][Validator] Fix apc cache service & config

### PR DESCRIPTION
**Pull request by ogizanagi**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #16793 |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

**Comment by jderusse** [comment]
Beware, the new service does not implements the same interface.
Your PR should break here https://github.com/ogizanagi/symfony/blob/ab4262e98a5121144e6d4e129b95e80b55ded08b/src/Symfony/Component/Validator/ValidatorBuilderInterface.php#L130

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -436,12 +436,7 @@ private function addValidationSection(ArrayNodeDefinition $rootNode)
                     ->info('validation configuration')
                     ->canBeEnabled()
                     ->children()
-                        ->scalarNode('cache')
-                            ->beforeNormalization()
-                                // Can be removed in 3.0, once ApcCache support is dropped
-                                ->ifString()->then(function ($v) { return 'apc' === $v ? 'validator.mapping.cache.apc' : $v; })
```
Comment (jderusse) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: This BC break should trigger a deprecation in previous versions right?

**Comment by ogizanagi** [comment]
Hmm, you're right. It should be based on `Symfony\Component\Validator\Mapping\Cache\DoctrineCache` which adapts it to this interface.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (ogizanagi) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php]: The deprecation seems to have been triggered through the new service in previous versions...
(see https://github.com/symfony/symfony/blob/2.8/src/Symfony/Bundle/FrameworkBundle/Resources/config/validator.xml#L42-L51)

And this service have been removed in 3.0.
I'm getting lost in the operation. I think something went wrong during a merge or this wasn't expected to be as it is right now.

I'm going to make `validator.mapping.cache.apc` an alias of the new `validator.mapping.cache.doctrine.apc` service.

**Comment by ogizanagi** [comment]
This PR has been submitted to `master` by mistake on `master`. (Bad habit, 3.0 is now released...)
Closing in favor of #16795.

**Pull request closed without merging**